types/src/transaction/authenticator.rs

```diff
@@ -300,6 +297,52 @@ impl TransactionAuthenticator {
             } => Some(fee_payer_signer.clone()),
         }
     }
+
+    pub fn all_signers(&self) -> Vec<AccountAuthenticator> {
```

Comment:
I think the other approach here might be to make a map function that can then test your lambda against each authenticator... but ... this is good enough